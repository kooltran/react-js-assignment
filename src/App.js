import React from "react";
import { connect } from "react-redux";
import { tripAction } from "./redux/tripAction";
import loadingGif from "./assets/loading.svg";
import { TripItem } from "./tripItem/tripItem";
import {
  StyledTripWrapper,
  StyledAppContainer,
  StyledAppTitle
} from "./App.styled";

export class App extends React.Component {
  componentDidMount() {
    this.props.tripAction();
  }

  render() {
    const { data, fetchState } = this.props;

    return (
      <StyledAppContainer>
        {fetchState === "loading" && <img src={loadingGif} alt="loading" />}
        {fetchState === "success" && (
          <StyledTripWrapper>
            <StyledAppTitle>Trips Management</StyledAppTitle>
            {data.trips.map(trip => (
              <TripItem key={trip.trip_id} tripData={trip} />
            ))}
          </StyledTripWrapper>
        )}
      </StyledAppContainer>
    );
  }
}

const mapStateToProps = state => {
  return {
    data: state.data,
    fetchState: state.fetchState
  };
};

const mapDispatchToProps = {
  tripAction
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
