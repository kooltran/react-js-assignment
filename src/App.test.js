import React from "react";
import { shallow } from "enzyme";
import { App } from "./App";

describe("renders App component", () => {
  const baseProps = {
    fetchState: "loading",
    tripAction: jest.fn(),
    data: {
      trips: [
        {
          distance: 12,
          duration: 123,
          start_time: "2000-11-22T03:44:55+0000",
          end_time: "2000-11-22T03:55:66+0000",
          trip_id: 10000
        },
        {
          distance: 13,
          duration: 456,
          start_time: "2000-11-22T03:44:55+0000",
          end_time: "2000-11-22T03:55:66+0000",
          trip_id: 1111
        }
      ]
    }
  };

  it("should render properly with loading state", () => {
    expect(shallow(<App {...baseProps} />)).toMatchSnapshot();
  });

  it("should dispatch an action on componentDidMount", () => {
    expect(baseProps.tripAction).toHaveBeenCalledTimes(1);
  });
});
