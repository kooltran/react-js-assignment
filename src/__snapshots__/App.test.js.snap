// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component should render properly with loading state 1`] = `ShallowWrapper {}`;
